[docs] Add guide for removing CloudPermanent node in VSphere (#20705)
deckhouse/deckhouse: docs/documentation/pages/admin/configuration/platform-scaling/node/CLOUD_NODE.md

Deckhouse Kubernetes Platform can run on top of Managed Kubernetes services, such as GKE and EKS. In this case, the [`node-manager`](/modules/node-manager/) module manages node configuration and automates node-related operations, but some features may be limited by the API of the corresponding cloud provider.

### Removing CloudPermanent nodes in vSphere

> Nodes of the [`CloudPermanent`](../../../../architecture/cluster-and-infrastructure/node-management/cloud-permanent-nodes.html) type are created based on the node group configuration specified in the [`nodeGroups`](/modules/cloud-provider-vsphere/cluster_configuration.html#vsphereclusterconfiguration-nodegroups) section of the `VsphereClusterConfiguration` resource.

Reviewer: ```suggestion
Nodes of the [`CloudPermanent`](../../../../architecture/cluster-and-infrastructure/node-management/cloud-permanent-nodes.html) type are created based on the node group configuration specified in the [`nodeGroups`](/modules/cloud-provider-vsphere/cluster_configuration.html#vsphereclusterconfiguration-nodegroups) section of the VsphereClusterConfiguration resource.
```

> The [`replicas`](/modules/cloud-provider-vsphere/cluster_configuration.html#vsphereclusterconfiguration-nodegroups-replicas) parameter defines the required number of virtual machines in the group. After the configuration is changed, the `dhctl converge` command runs Terraform and brings the state of virtual machines in VMware vSphere in line with the specified number of replicas.

Reviewer: ```suggestion
The [`replicas`](/modules/cloud-provider-vsphere/cluster_configuration.html#vsphereclusterconfiguration-nodegroups-replicas) parameter defines the required number of virtual machines in the group. After the configuration is changed, run the `dhctl converge` command to start Terraform and bring the state of virtual machines in VMware vSphere in line with the specified number of replicas.
```

To reduce the number of nodes in a group, decrease the `replicas` value and run `dhctl converge`.

{% alert level="warning" %}
Do not remove the group definition from the [`nodeGroups`](/modules/cloud-provider-vsphere/cluster_configuration.html#vsphereclusterconfiguration-nodegroups) section while the value of the [`replicas`](/modules/cloud-provider-vsphere/cluster_configuration.html#vsphereclusterconfiguration-nodegroups-replicas) parameter is greater than zero.

If the group definition is removed before the number of replicas is reduced to zero, the Terraform state may become inconsistent with the state of virtual machines and disks in VMware vSphere. As a result, a subsequent `dhctl converge` run may fail and require manual state recovery.
{% endalert %}

#### Reducing the number of nodes

To reduce the number of nodes in a `CloudPermanent` group:

> 1. Open the cloud provider configuration for editing:

Reviewer: ```suggestion
1. Open the vSphere configuration for editing:
```

   ```shell
   d8 system edit provider-cluster-configuration
   ```

1. In the [`nodeGroups`](/modules/cloud-provider-vsphere/cluster_configuration.html#vsphereclusterconfiguration-nodegroups) section, find the required group and decrease the value of the [`replicas`](/modules/cloud-provider-vsphere/cluster_configuration.html#vsphereclusterconfiguration-nodegroups-replicas) parameter to the required number of nodes.

   For example, to reduce the number of nodes in the `worker` group from three to two, set `replicas: 2`:

   ```yaml
   nodeGroups:
   - name: worker
     replicas: 2
     zones:
     - zone-a
     instanceClass:
       numCPUs: 4
       memory: 8192
       template: Templates/ubuntu-24.04
       datastore: datastore-1
       mainNetwork: network-1
   ```

   Save the changes.

1. In the DKP installer container, apply the updated configuration:

   ```shell
   dhctl converge \
     --ssh-host <MASTER-NODE-IP-ADDRESS> \
     --ssh-user <USERNAME> \
     --ssh-agent-private-keys /tmp/.ssh/<PRIVATE-SSH-KEY-NAME>
   ```

>    > Use an installer container of the same edition and version as the cluster.

Reviewer: ```suggestion
   {% alert level="info" %}
   Use an installer container of the same edition and version as the cluster.
   {% endalert %}
```

1. Wait for `dhctl converge` to complete and check the number of nodes in the group:

   ```shell
   d8 k get nodes -l node.deckhouse.io/group=<GROUP-NAME>
   ```

#### Completely removing a node group

> A `CloudPermanent` group must be removed in two stages. First, reduce the number of replicas to zero and wait for the nodes and virtual machines to be removed. You can then remove the group definition from the cloud provider configuration.

Reviewer: ```suggestion
A `CloudPermanent` group must be removed in two stages. First, reduce the number of replicas to zero and wait for the nodes and virtual machines to be removed. You can then remove the group definition from the vSphere configuration.
```

To completely remove a `CloudPermanent` group:

> 1. Open the cloud provider configuration for editing:

Reviewer: ```suggestion
1. Open the vSphere configuration for editing:
```

   ```yaml
   nodeGroups:
   - name: worker
     replicas: 0
     zones:
     - zone-a
     instanceClass:
       numCPUs: 4
       memory: 8192
       template: Templates/ubuntu-24.04
       datastore: datastore-1
       mainNetwork: network-1
   ```

   Save the changes.

1. In the DKP installer container, apply the updated configuration:

   ```shell
   dhctl converge \
     --ssh-host <MASTER-NODE-IP-ADDRESS> \
     --ssh-user <USERNAME> \
     --ssh-agent-private-keys /tmp/.ssh/<PRIVATE-SSH-KEY-NAME>
   ```

1. Wait for `dhctl converge` to complete successfully and make sure that no nodes remain in the group:

   ```shell
   d8 k get nodes -l node.deckhouse.io/group=<GROUP-NAME>
   ```

   The command must not return any nodes from the group being removed.

   Also use vSphere Client to make sure that the virtual machines associated with the group have been removed.

1. Open the cloud provider configuration again:

   ```shell
   d8 system edit provider-cluster-configuration
   ```

> 1. Remove the group definition from the `nodeGroups` section, save the changes, and close the editor.

Reviewer: ```suggestion
1. Remove the group definition from the `nodeGroups` section and save the changes.
```
